### PR TITLE
8th pull request

### PR DESCRIPTION
**Pull request by sawkiewicz-tomasz**

### CODE REVIEW

File: src/main/java/sawkiewicz/task1/Answer.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    public Answer(String responce, String message) {
+        this.responce = responce;
+        this.message = message;
+    }
+
+    public Answer() {
+    }
+
+    public void setResponce(String responce) {
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/Answer.java:15]: remove

File: src/main/java/sawkiewicz/task1/Answer.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        this.responce = responce;
+    }
+
+    public void setMessage(String message) {
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/Answer.java:19]: remove

File: src/main/java/sawkiewicz/task1/Answer.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+public class Answer {
+    private String responce;
+    private String message;
+
+    public Answer(String responce, String message) {
+        this.responce = responce;
+        this.message = message;
+    }
+
+    public Answer() {
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/Answer.java:12]: remove

File: src/main/java/sawkiewicz/task1/CartItem.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        this.item = item;
+    }
+
+    public CartItem() {
+    }
+
+    public void increaseQuantiy (CartItem cartItem){
+        int temp = cartItem.getQuantity();
+        temp++;
+        cartItem.setQuantity(temp);
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/CartItem.java:30]: `cartItem.quantity = temp;`

File: src/main/java/sawkiewicz/task1/CartItem.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package sawkiewicz.task1;
+
+public class CartItem {
+    private int quantity;
+    private Item item;
+
+    public void setQuantity(int quantity) {
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/CartItem.java:7]: remove

File: src/main/java/sawkiewicz/task1/Repository.java [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package sawkiewicz.task1;
+
+public interface Repository <K>{
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/Repository.java:3]:
````
interface Repository<K, V>

   V findById(K id)
```

File: src/main/java/sawkiewicz/task1/ItemRepository.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
+package sawkiewicz.task1;
+
+import com.google.common.collect.ImmutableList;
+import java.util.ArrayList;
+import java.util.List;
+import static java.util.stream.Collectors.toCollection;
+
+public class ItemRepository<String> implements Repository {
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/ItemRepository.java:8]: `ItemRepository implements Repository<String, Item>`

File: src/main/java/sawkiewicz/task1/ShoppingCart.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
+package sawkiewicz.task1;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ShoppingCart {
+    private String id;
+    private List <CartItem> cartItems = new ArrayList<>();
+
+    public ShoppingCart() {
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/ShoppingCart.java:10]: this is default constructor, no need to write it

File: src/main/java/sawkiewicz/task1/ShoppingCartService.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class ShoppingCartService {
+
+    @Autowired
+    ShoppingCartRepository shoppingCartRepository;
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/ShoppingCartService.java:11]: private

File: src/main/java/sawkiewicz/task1/ShoppingCartService.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+    @Autowired
+    ItemRepository itemRepository;
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/ShoppingCartService.java:14]: private

File: src/main/java/sawkiewicz/task1/ShoppingCartService.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+    public ShoppingCartService(ShoppingCartRepository shoppingCartRepository) {//, ItemRepository itemRepository) {
+        this.shoppingCartRepository = shoppingCartRepository;
+    }
+
+    public void addItemToCart(String cartId, Item itemToAdd) {
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/ShoppingCartService.java:20]: cover with unit tests !!

File: src/main/java/sawkiewicz/task1/ShoppingCartService.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+public class ShoppingCartService {
+
+    @Autowired
+    ShoppingCartRepository shoppingCartRepository;
+
+    @Autowired
+    ItemRepository itemRepository;
+
+    public ShoppingCartService(ShoppingCartRepository shoppingCartRepository) {//, ItemRepository itemRepository) {
+        this.shoppingCartRepository = shoppingCartRepository;
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/ShoppingCartService.java:17]: use Mockito mock instead of real repository in JUnit tests

Comment (marcindziedzic) [reply src/main/java/sawkiewicz/task1/ShoppingCartService.java:17]: http://site.mockito.org/

File: src/main/java/sawkiewicz/task1/Task1Controller.java [hunk 13]
```diff
@@ -0,0 +1,43 @@
+package sawkiewicz.task1;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+
+@RestController
+public class Task1Controller {
+    ItemRepository itemRepository = new ItemRepository();
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/Task1Controller.java:9]: private !

File: src/main/java/sawkiewicz/task1/Task1Controller.java [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+    ShoppingCartRepository shoppingCartRepository = new ShoppingCartRepository();
+    ShoppingCartService shoppingCartService = new ShoppingCartService(shoppingCartRepository);
+
+    @RequestMapping(method = RequestMethod.GET, value = "/carts/{cartId}/price")
+    public ResponseEntity<Answer>showTotalPrice(@PathVariable String cartId) {
+        Answer answer = shoppingCartService.showPrice(cartId);
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/Task1Controller.java:15]: `Response` not `Answer`

File: src/main/java/sawkiewicz/task1/Task1Controller.java [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.List;
+
+@RestController
+public class Task1Controller {
+    ItemRepository itemRepository = new ItemRepository();
+    ShoppingCartRepository shoppingCartRepository = new ShoppingCartRepository();
+    ShoppingCartService shoppingCartService = new ShoppingCartService(shoppingCartRepository);
+
+    @RequestMapping(method = RequestMethod.GET, value = "/carts/{cartId}/price")
+    public ResponseEntity<Answer>showTotalPrice(@PathVariable String cartId) {
```
Comment (marcindziedzic) [inline src/main/java/sawkiewicz/task1/Task1Controller.java:14]: `public Answer showTotalPrice`

File: src/main/java/sawkiewicz/task1/Repository.java (hunk 6 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/Repository.java:3]: Done

File: src/main/java/sawkiewicz/task1/Answer.java (hunk 3 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/Answer.java:12]: removed

File: src/main/java/sawkiewicz/task1/Answer.java (hunk 1 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/Answer.java:15]: removed

File: src/main/java/sawkiewicz/task1/Answer.java (hunk 2 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/Answer.java:19]: removed

File: src/main/java/sawkiewicz/task1/CartItem.java (hunk 4 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/CartItem.java:30]: done

File: src/main/java/sawkiewicz/task1/ShoppingCart.java (hunk 8 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/ShoppingCart.java:10]: removed

File: src/main/java/sawkiewicz/task1/ShoppingCartService.java (hunk 9 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/ShoppingCartService.java:11]: done

File: src/main/java/sawkiewicz/task1/ShoppingCartService.java (hunk 10 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/ShoppingCartService.java:14]: done

File: src/main/java/sawkiewicz/task1/Task1Controller.java (hunk 13 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/Task1Controller.java:9]: done

File: src/main/java/sawkiewicz/task1/Task1Controller.java (hunk 14 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/Task1Controller.java:15]: renamed

File: src/main/java/sawkiewicz/task1/Task1Controller.java (hunk 15 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/Task1Controller.java:14]: changed

File: src/main/java/sawkiewicz/task1/CartItem.java (hunk 5 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/CartItem.java:7]: removed

File: src/main/java/sawkiewicz/task1/ItemRepository.java (hunk 7 above)
Comment (sawkiewicz-tomasz) [reply src/main/java/sawkiewicz/task1/ItemRepository.java:8]: done

**Pull request closed without merging**